### PR TITLE
filter our non-printable characters

### PR DESCRIPTION
**Pull request by kai5263499**
Small change to the BytesWritable toString method to filter out non-printable characters so that chr doesn't throw an exception

### CODE REVIEW

File: python-hadoop/hadoop/io/BytesWritable.py [hunk 1]
```diff
@@ -28,4 +28,4 @@ def readFields(self, data_input):
         self._value = data_input.readFully(size)
 
     def toString(self):
-        return ''.join(chr(x) for x in self._value)
+        return ''.join(chr(x) for x in self._value if x > 31 or x == 9)
```
Comment (cscheffler) [inline python-hadoop/hadoop/io/BytesWritable.py]: I also ran into this problem and this works even better:

    return ''.join(chr(x % 256) for x in self._value)

The main problem is that values are signed short integers (in the range -128 to 127) and `chr` is expecting unsigned integers (in the range 0 to 255).

Would you please confirm that this more general solution works for you too, and update the PR?

Thanks,
Carl

Comment (kai5263499) [inline python-hadoop/hadoop/io/BytesWritable.py]: The mod 256 method adds some unprintable characters. Mod 128 doesn't, though. I suppose it really depends on what the goal of toString is. If our goal is to not blow up when making a best effort to stringify binary data then I agree that the mod 255 method fits the bill

Comment (cscheffler) [inline python-hadoop/hadoop/io/BytesWritable.py]: That is true, but I don't think there is a requirement that characters be printable. My understanding is that you could have a byte string of any kind. The mod 256 does seem to take care of the exceptions and is more general.

**Comment by kai5263499** [comment]
Ok, I've changed it to use the mod 256 method

**Comment by cscheffler** [comment]
Awesome, thanks!

**Comment by meneldor1** [comment]
FWIW this patch helped me read a SequenceFile containing protobuf messages

**Pull request merged**